runtime/src/main/java/org/corfudb/runtime/MultiCheckpointWriter.java

```diff
@@ -71,14 +72,27 @@ public Token appendCheckpoints(CorfuRuntime rt, String author) {
                                 .getSerializer();
                 cpw.setSerializer(serializer);
 
-                Token minCPSnapshot = cpw.appendCheckpoint();
+                Token minCPSnapshot = Token.UNINITIALIZED;
```

Comment:
Since the snapshot returned `appendCheckpoint` is a global snapshot that shouldn't regress, shouldn't we just select the first checkpoints snapshot and make sure subsequent checkpoints all return greater snapshots (instead of just finding a min of all snapshots).

Reply:
+1

Reply:
Sure, then I'll throw an exception if by chance a snapshot is lower cause that would mean there was some kind of regression.